### ISSUE TITLE
Changeset diff against the remote branch

### ISSUE THREAD

**Issue by timotheeguerin** [changesets/changesets#1299]
## Affected Packages

@changeset/cli
@changeset/git

## Problem

When you run `changeset add` locally it will do a diff against your local main branch instead of the current remote but if you updated your local working branch with the remote main branch without updating your local main branch (e.g. `git pull origin main`) then it will report changes in other packages(from any commits made into main since).

## Proposed solution

It would be nice to have a way to resolve the diff against the remote branch so you don't have to keep your local base branch up to date.

**Comment by bluwy** [comment]
I don't think we can easily do that due to git limitations. The only way we can diff is to first fetch updates locally, which I don't think changesets shouldn't be doing automatically as you may not want that / intentionally working on an older update.

The other alternative is to fetch the branch to a separate changesets-managed branch, like `changesets-diff/main` or something, but I think it may also be surprising that changesets is creating these branches on behalf. I'll leave this open for now though if others have thoughts on supporting this.

**Comment by beeequeue** [comment]
does it make sense to check against the remote branch? this only matters if the local branch is behind (or ahead *and* behind), so when would the changed packages be different in a way that is helpful when comparing against the remote vs. locally?

**Comment by Andarist** [comment]
It seems this would kinda work without changing any local refs:
```
git fetch origin refs/heads/main
git merge-base HEAD FETCH_HEAD
git diff FETCH_HEAD...HEAD
```

> does it make sense to check against the remote branch? 

The current branch could be synced with new version of the base branch on the remote and that could be fetched locally without having either `main` and `origin/main` synced with the remote state. So the current local branch could contain unrelated work that is not truly owned by it
